spec/policies/access_request_policy_spec.rb

```diff
@@ -18,4 +18,20 @@
       it { should permit(user, access_request) }
     end
   end
+
+  permissions :index? do
+    let(:access_request) { build(:access_request) }
+
+    context 'non-admin user' do
+      let(:user) { build(:user) }
+
+      it { should_not permit(user, access_request) }
```

Comment:
I don't think you need the `access_request` here. You should be able to just do 

`it { should_not permit(user) }`

That's because the `index?` method doesn't need an access request - we only need the user to see if it can get the list of access requests.

Reply:
Separately, if the `index?` and `approve?` would be exactly the same, you can do `permissions :approve?, :index? do ...`

See https://github.com/varvet/pundit#policy-specs

Just figured I'd share, I had forgotten myself :D

Reply:
> That's because the index? method doesn't need an access request

I think it needs `AccessRequest` since technically we're checking the user against the model class, and not (as you say) an individual access request.

Reply:
hmm I thought that was just for the magic in `authorize` to work

Reply:
Just noticed that we alias `index?` to `approve?` so maybe let's test these the way I suggested before:

`permissions :approve?, :index? do ...`

By modifying the test for `approve?` that is. Then we can get rid of this one.

Reply:
Cool. Done. Thanks guys

Reply:
Spoke in-person, makes sense to me assuming `AccessRequest` really isn't needed.